src/promises/allNoLeaks.ts

```diff
@@ -0,0 +1,30 @@
+// Inspired by https://stackoverflow.com/questions/64928212/how-to-use-promise-allsettled-with-typescript#69451755
+const isRejected = (result: PromiseSettledResult<unknown>) =>
```

Comment:
We don't need `<unknown>` here

Reply:
OK, can you help with the resulting lint errors then?

Reply:
Yeah fancy those!  Clearly I'm wrong  regarding the `<unknown>` being assumed.

I am now curious though so I'm gonna spend just 5m learning